dynamiqs/utils/states.py

```diff
@@ -34,6 +34,15 @@ def fock(
                 [0.+0.j],
                 [0.+0.j],
                 [0.+0.j]], dtype=torch.complex128)
+
+    Args:
+        dims: Dimension of the Hilbert space of each mode.
+        states: Fock state of each mode.
+        dtype: Data type of the returned tensor.
+        device: Device of the returned tensor.
+
+    Returns:
+        torch.Tensor (n, 1): Ket of the Fock state or tensor product of Fock states.
```

Comment:
```suggestion
    (n, 1): Ket of the Fock state or tensor product of Fock states.
```

Reply:
I should clarify how I think we should handle types:
- By default: in the code, we only write the type in the function signature, never in the docstring. In the doc website, the types only appear in the function signature.
- IF one of the argument or the returned object is a tensor, in the code we add the tensor dimension to the dosctring between parentheses. In the doc website it appears in the `Args` or `Returns` section next to the argument name.
- IF we want to write a custom type description because the function signature is a bit obscure (like `OperatorLike | List[OperatorLike] | None`), in the code we add the custom type description to the dosctring between parentheses. In the doc website it appears in the `Args` or `Returns` section next to the argument name.

Note that the doc is a bit bugged now, because
- The function signature is not always correct.
- The typing also appears in the `Args` section even if we didn't specify a custom type in the docstring.

What do you think of these conventions? It's fine for this PR, I'll fix the doc after we've merged.